drivers/flash/nrf_qspi_nor.c

```diff
@@ -79,6 +79,16 @@ BUILD_ASSERT(((INST_0_QER == JESD216_DW15_QER_NONE)
 	      || (INST_0_QER == JESD216_DW15_QER_S1B6)),
 	     "Driver only supports NONE or S1B6 for quad-enable-requirements");
 
+#define INST_0_4BA DT_INST_PROP_OR(0, enter_4byte_addr, 0)
+#if (INST_0_4BA != 0)
+BUILD_ASSERT(((INST_0_4BA & 0x03) != 0),
+	     "Driver only supports command (0xB7) for entering 4 byte addressing mode");
+BUILD_ASSERT((DT_INST_PROP(0, address_size_32) == NRF_QSPI_ADDRMODE_32BIT),
```

Comment:
```suggestion
BUILD_ASSERT(DT_INST_PROP(0, address_size_32),
```
This is a boolean property, so it should not be compared with the value of `NRF_QSPI_ADDRMODE_32BIT` (which by chance evaluates to 1, hence it works in the current form).